### PR TITLE
42 build new cli

### PR DESCRIPTION
**Pull request by ductiletoaster**
Completed the first part of the CLI upgrade. Using an entirely new framework so we can build CLI functionality much much faster.

https://github.com/infinitered/gluegun

For this version I am not focusing on testing the CLI interface since it mostly uses already tested functions and scripts. However I would like to explore a more rigourous integration test setup for v6.

### CODE REVIEW

**Comment by ductiletoaster** [comment]
Still need to port over webpack build and dev server + storybook. Then I'd like to fully document the new CLI

File: examples/react-ssr-example/package.json [hunk 1]
```diff
@@ -84,31 +82,30 @@
...
     "env:production": "cp .env.localhost .env",
     "generate": "pixeloven-generators generate",
     "lint": "run-p lint:ts lint:scss",
-    "lint:ts": "pixeloven-tasks lint:ts src/**/*.{ts,tsx}",
-    "lint:scss": "pixeloven-tasks lint:scss src/**/*.{scss}",
-    "pretest": "pixeloven-tasks test:clean",
-    "pretty": "pixeloven-tasks pretty src/**/*.{scss,ts,tsx}",
-    "pretty:ts": "pixeloven-tasks pretty:ts src/**/*.{ts,tsx}",
-    "pretty:scss": "pixeloven-tasks pretty:scss src/**/*.{scss}",
+    "lint:ts": "pixeloven lint ts ./src/**/*.{ts,tsx}",
```
Comment (jimmynotjim) [inline examples/react-ssr-example/package.json]: Should we think about an `@root` shorthand for stuff like this?

File: packages/pixeloven-cli/src/commands/pretty.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+                    print.error(`Prettier exited with status ${statusCode}`);
+                    break;
+                }
+                print.success(`Success! Looks a lot nicer now doesn't it?!`);
+                break;
+            case "ts":
+            case "tsx":
+                statusCode = await tsLint(["--fix"].concat(argList));
+                if (statusCode) {
+                    print.error(`Tslint exited with status ${statusCode}`);
```
Comment (jimmynotjim) [inline packages/pixeloven-cli/src/commands/pretty.ts]: Nit: the product name is `TSLint` 

:pedantic:

File: packages/pixeloven-cli/src/commands/pretty.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+                    break;
+                }
+                statusCode = await prettier(argList);
+                if (statusCode) {
+                    print.error(`Prettier exited with status ${statusCode}`);
+                    break;
+                }
+                print.success(`Success! Looks a lot nicer now doesn't it?!`);
```
Comment (jimmynotjim) [inline packages/pixeloven-cli/src/commands/pretty.ts]: Ha! I love this.

File: packages/pixeloven-cli/src/extensions/prettier.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import { PixelOvenRunContext } from "../types";
```
Comment (jimmynotjim) [inline packages/pixeloven-cli/src/extensions/prettier.ts]: These three files are 90% the same. Any reason not to combine them to one and accept arguments for the differences?

File: packages/pixeloven-webpack/prettier.json [hunk 5]
```diff
@@ -0,0 +1,4 @@
+{
```
Comment (jimmynotjim) [inline packages/pixeloven-webpack/prettier.json]: Is it worth moving these prettier rules to the root and extending them like you've done with tsconfigs?

File: packages/pixeloven-cli/src/commands/pretty.ts (hunk 3 above)
Comment (ductiletoaster) [reply packages/pixeloven-cli/src/commands/pretty.ts]: wanted more character! Let me know if you got some better phrases!

File: examples/react-ssr-example/package.json (hunk 1 above)
Comment (ductiletoaster) [reply examples/react-ssr-example/package.json]: I don't believe those will work in this CLI context. We luckily support globbing in this pattern which automatically gets parsed into individual file paths matching that pattern

File: packages/pixeloven-webpack/prettier.json (hunk 5 above)
Comment (ductiletoaster) [reply packages/pixeloven-webpack/prettier.json]: Yeah I'd like to do that just never got around to it.

File: packages/pixeloven-cli/src/extensions/prettier.ts (hunk 4 above)
Comment (ductiletoaster) [reply packages/pixeloven-cli/src/extensions/prettier.ts]: yeah I'll like do some cleanup once I figure out how to do tests and get parity with the old -tasks deal. Good call out!

**Pull request merged**